diff --git a/packages-internal/middleware-sdk-route53/src/middleware-sdk-route53.integ.spec.ts b/packages-internal/middleware-sdk-route53/src/middleware-sdk-route53.integ.spec.ts
index 143e42412365..97eb38b49424 100644
--- a/packages-internal/middleware-sdk-route53/src/middleware-sdk-route53.integ.spec.ts
+++ b/packages-internal/middleware-sdk-route53/src/middleware-sdk-route53.integ.spec.ts
@@ -49,7 +49,7 @@ describe("middleware-sdk-route53", () => {
       });
 
       requireRequestsFrom(client).toMatch({
-        path: "/2013-04-01/change/my-change",
+        path: /^\/20\d\d-\d\d-\d\d\/change\/my\-change$/,
       });
 
       await client.getChange({
