diff --git a/apps/badgeuser/api.py b/apps/badgeuser/api.py
index 0c3cf2ee..37033dbf 100644
--- a/apps/badgeuser/api.py
+++ b/apps/badgeuser/api.py
@@ -120,6 +120,7 @@ class AcceptTermsView(APIView):
     """
     Endpoint used for accepting terms
     POST to accept terms
+    DELETE to reject terms
     """
     model = Terms
     permission_classes = (AuthenticatedWithVerifiedEmail,)
@@ -153,7 +154,10 @@ def post(self, request, **kwargs):
     )
     def delete(self, request, **kwargs):
         if request.data:
-            term_agreement = TermsAgreement.objects.get(entity_id=request.data['terms_agreement_entity_id'])
+            term_agreement = TermsAgreement.objects.get(
+                user=request.user,
+                entity_id=request.data['terms_agreement_entity_id'],
+            )
             term_agreement.agreed = False
             term_agreement.save()
             request.user.remove_cached_data(['cached_terms_agreements'])
diff --git a/apps/issuer/api.py b/apps/issuer/api.py
index 988aa7ec..714f938f 100644
--- a/apps/issuer/api.py
+++ b/apps/issuer/api.py
@@ -9,6 +9,7 @@
 import badgrlog
 from entity.api import BaseEntityListView, BaseEntityDetailView, VersionedObjectMixin, BaseEntityView, BaseArchiveView
 from issuer.models import Issuer, BadgeClass, BadgeInstance, BadgeInstanceCollection
+from issuer.permissions import IsBadgeInstanceOwner
 from issuer.serializers import (
     IssuerSerializer,
     BadgeClassSerializer,
@@ -197,7 +198,7 @@ def post(self, request, **kwargs):
 
 class BadgeInstanceCollectionDetail(BaseEntityDetailView):
     model = BadgeInstanceCollection
-    permission_classes = (AuthenticatedWithVerifiedEmail,)
+    permission_classes = (AuthenticatedWithVerifiedEmail, IsBadgeInstanceOwner)
     serializer_class = BadgeInstanceCollectionSerializer
     http_method_names = ['delete', 'put']
 
diff --git a/apps/issuer/permissions.py b/apps/issuer/permissions.py
index 2728dd5c..06ed6b8d 100644
--- a/apps/issuer/permissions.py
+++ b/apps/issuer/permissions.py
@@ -79,3 +79,12 @@ def valid_scopes_for_view(cls, view, method=None):
             return []
 
         return valid_scopes
+
+
+class IsBadgeInstanceOwner(permissions.BasePermission):
+    """
+    Only grant access to owner of BadgeInstance.
+    """
+
+    def has_object_permission(self, request, view, obj):
+        return obj.user == request.user
diff --git a/apps/issuer/serializers.py b/apps/issuer/serializers.py
index fadeb8c9..b6a76720 100644
--- a/apps/issuer/serializers.py
+++ b/apps/issuer/serializers.py
@@ -643,7 +643,7 @@ def update(self, instance, validated_data):
 
     def validate_badge_instances(self, badge_instances):
         user = self.context['request'].user
-        for bc in badge_instances:
-            if bc.user != user:
+        for badge_instance in badge_instances:
+            if badge_instance.user != user:
                 raise IntegrityError('BadgeInstance must be owned by the current user.')
         return badge_instances
