diff --git a/server/jetstream_api.go b/server/jetstream_api.go
index 826ca5f61d0..937af6f7c88 100644
--- a/server/jetstream_api.go
+++ b/server/jetstream_api.go
@@ -2395,6 +2395,10 @@ func (s *Server) jsConsumerLeaderStepDownRequest(sub *subscription, c *client, _
 
 	js.mu.RLock()
 	isLeader, sa := cc.isLeader(), js.streamAssignment(acc.Name, stream)
+	var ca *consumerAssignment
+	if sa != nil {
+		ca = sa.consumers[consumer]
+	}
 	js.mu.RUnlock()
 
 	if isLeader && sa == nil {
@@ -2411,10 +2415,6 @@ func (s *Server) jsConsumerLeaderStepDownRequest(sub *subscription, c *client, _
 		return
 	}
 
-	var ca *consumerAssignment
-	if sa.consumers != nil {
-		ca = sa.consumers[consumer]
-	}
 	if ca == nil {
 		resp.Error = NewJSConsumerNotFoundError()
 		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
