diff --git a/pdns/auth-main.cc b/pdns/auth-main.cc
index 2866084f91a5..e07591fb71a4 100644
--- a/pdns/auth-main.cc
+++ b/pdns/auth-main.cc
@@ -856,7 +856,9 @@ static void mainthread()
     DNSPacket::s_doEDNSCookieProcessing = true;
     std::vector<std::string> secrets;
     stringtok(secrets, ::arg()["edns-cookie-secret"], ",");
-    DNSPacket::s_OldEDNSCookieKeys.reserve(secrets.size() - 1);
+    auto lockedEDNSCookieKey = DNSPacket::s_EDNSCookieKey.lock();
+    auto lockedOldEDNSCookieKeys = DNSPacket::s_OldEDNSCookieKeys.lock();
+    lockedOldEDNSCookieKeys->reserve(secrets.size() - 1);
 
     bool first{true};
     for (const auto& secret : secrets) {
@@ -868,7 +870,7 @@ static void mainthread()
         }
         std::array<char, EDNSCookiesOpt::EDNSCookieSecretSize / 2> key{};
         dns_random(key.data(), key.size());
-        DNSPacket::s_EDNSCookieKey = std::string(key.data(), key.size());
+        *lockedEDNSCookieKey = std::string(key.data(), key.size());
       }
       else {
         try {
@@ -877,10 +879,10 @@ static void mainthread()
           }
           auto secretBytes = makeBytesFromHex(secret);
           if (first) {
-            DNSPacket::s_EDNSCookieKey = std::move(secretBytes);
+            *lockedEDNSCookieKey = std::move(secretBytes);
           }
           else {
-            DNSPacket::s_OldEDNSCookieKeys.emplace_back(std::move(secretBytes));
+            lockedOldEDNSCookieKeys->emplace_back(std::move(secretBytes));
           }
         }
         catch (const std::range_error& e) {
diff --git a/pdns/dnspacket.cc b/pdns/dnspacket.cc
index bfefdc8f10ea..6f6007506bea 100644
--- a/pdns/dnspacket.cc
+++ b/pdns/dnspacket.cc
@@ -56,8 +56,8 @@
 
 bool DNSPacket::s_doEDNSSubnetProcessing;
 bool DNSPacket::s_doEDNSCookieProcessing;
-string DNSPacket::s_EDNSCookieKey;
-std::vector<std::string> DNSPacket::s_OldEDNSCookieKeys;
+LockGuarded<string> DNSPacket::s_EDNSCookieKey;
+LockGuarded<std::vector<std::string>> DNSPacket::s_OldEDNSCookieKeys;
 uint16_t DNSPacket::s_udpTruncationThreshold;
 
 DNSPacket::DNSPacket(Logr::log_t slog, bool isQuery): d_isQuery(isQuery), d_slog(slog)
@@ -354,7 +354,7 @@ void DNSPacket::wrapup(bool throwsOnTruncation)
       }
 
       if (d_haveednscookie && d_eco.isWellFormed()) {
-        d_eco.makeServerCookie(s_EDNSCookieKey, getInnerRemote());
+        d_eco.makeServerCookie(*s_EDNSCookieKey.read_only_lock(), getInnerRemote());
         opts.emplace_back(EDNSOptionCode::COOKIE, d_eco.makeOptString());
       }
 
@@ -631,7 +631,7 @@ try
       else if (s_doEDNSCookieProcessing && option.first == EDNSOptionCode::COOKIE) {
         d_haveednscookie = true;
         d_eco.makeFromString(option.second);
-        d_ednscookievalid = d_eco.isValid(s_EDNSCookieKey, getInnerRemote(), s_OldEDNSCookieKeys);
+        d_ednscookievalid = d_eco.isValid(*s_EDNSCookieKey.read_only_lock(), getInnerRemote(), *s_OldEDNSCookieKeys.read_only_lock());
       }
       else {
         // cerr<<"Have an option #"<<iter->first<<": "<<makeHexDump(iter->second)<<endl;
diff --git a/pdns/dnspacket.hh b/pdns/dnspacket.hh
index 9d4ac481870a..b4e1e281264d 100644
--- a/pdns/dnspacket.hh
+++ b/pdns/dnspacket.hh
@@ -170,8 +170,8 @@ public:
   static uint16_t s_udpTruncationThreshold;
   static bool s_doEDNSSubnetProcessing;
   static bool s_doEDNSCookieProcessing;
-  static string s_EDNSCookieKey;
-  static std::vector<std::string> s_OldEDNSCookieKeys;
+  static LockGuarded<string> s_EDNSCookieKey;
+  static LockGuarded<std::vector<std::string>> s_OldEDNSCookieKeys;
   EDNSSubnetOpts d_eso;
 
 #ifdef ENABLE_GSS_TSIG
diff --git a/pdns/dynhandler.cc b/pdns/dynhandler.cc
index 321eb96ef50c..cdd8b05fcf29 100644
--- a/pdns/dynhandler.cc
+++ b/pdns/dynhandler.cc
@@ -129,33 +129,36 @@ string DLManageCookieSecret(const vector<string>& parts, [[maybe_unused]] pid_t
     return cookieUsage;
   }
 
+  auto lockedEDNSCookieKey = DNSPacket::s_EDNSCookieKey.lock();
+  auto lockedOldEDNSCookieKeys = DNSPacket::s_OldEDNSCookieKeys.lock();
+
   const auto &command = parts.at(1);
   if (command == "add") {
     if (parts.size() != 3) {
       return "Usage: cookie-secret add <SECRET|random>";
     }
-    auto oldSecret = DNSPacket::s_EDNSCookieKey;
+    auto oldSecret = *lockedEDNSCookieKey;
     std::array<char, EDNSCookiesOpt::EDNSCookieSecretSize / 2> key{};
     const auto &newSecret = parts.at(2);
     if (newSecret == "random") {
       dns_random(key.data(), key.size());
-      DNSPacket::s_EDNSCookieKey = std::string(key.data(), key.size());
+      *lockedEDNSCookieKey = std::string(key.data(), key.size());
     } else {
       if (newSecret.size() != EDNSCookiesOpt::EDNSCookieSecretSize) {
         return "wrong size for new secret (" + std::to_string(newSecret.size()) + "), must be " + std::to_string(EDNSCookiesOpt::EDNSCookieSecretSize);
       }
       try {
         auto secretBytes = makeBytesFromHex(newSecret);
-        DNSPacket::s_EDNSCookieKey = std::move(secretBytes);
+        *lockedEDNSCookieKey = std::move(secretBytes);
       } catch (const std::exception& exc) {
         return "Can not convert " + newSecret + " to bytes: " + exc.what();
       }
     }
     if (!oldSecret.empty()) {
-      DNSPacket::s_OldEDNSCookieKeys.insert(DNSPacket::s_OldEDNSCookieKeys.begin(), oldSecret);
+      lockedOldEDNSCookieKeys->insert(lockedOldEDNSCookieKeys->begin(), oldSecret);
     }
     DNSPacket::s_doEDNSCookieProcessing = true;
-    return "COOKIE secret set to " + makeHexDump(DNSPacket::s_EDNSCookieKey, "");
+    return "COOKIE secret set to " + makeHexDump(*lockedEDNSCookieKey, "");
   }
 
   if (command == "delete") {
@@ -164,11 +167,11 @@ string DLManageCookieSecret(const vector<string>& parts, [[maybe_unused]] pid_t
     }
     const auto &secret = parts.at(2);
     if (secret == "last") {
-      if (DNSPacket::s_OldEDNSCookieKeys.empty()) {
+      if (lockedOldEDNSCookieKeys->empty()) {
         return "No inactive keys, nothing removed";
       }
-      auto removedSecret = DNSPacket::s_OldEDNSCookieKeys.back();
-      DNSPacket::s_OldEDNSCookieKeys.pop_back();
+      auto removedSecret = lockedOldEDNSCookieKeys->back();
+      lockedOldEDNSCookieKeys->pop_back();
       return "COOKIE secret " + makeHexDump(removedSecret, "") +" removed";
     }
     if (secret.size() != EDNSCookiesOpt::EDNSCookieSecretSize) {
@@ -176,12 +179,12 @@ string DLManageCookieSecret(const vector<string>& parts, [[maybe_unused]] pid_t
     }
     try {
       auto secretBytes = makeBytesFromHex(secret);
-      if (secretBytes == DNSPacket::s_EDNSCookieKey) {
+      if (secretBytes == *lockedEDNSCookieKey) {
         return "COOKIE secret " + makeHexDump(secretBytes, "") +" is the active secret, refusing to remove";
       }
-      auto it = std::find(DNSPacket::s_OldEDNSCookieKeys.begin(), DNSPacket::s_OldEDNSCookieKeys.end(), secretBytes); // NOLINT(readability-identifier-length)
-      if (it != DNSPacket::s_OldEDNSCookieKeys.end()) {
-        DNSPacket::s_OldEDNSCookieKeys.erase(it);
+      auto it = std::find(lockedOldEDNSCookieKeys->begin(), lockedOldEDNSCookieKeys->end(), secretBytes); // NOLINT(readability-identifier-length)
+      if (it != lockedOldEDNSCookieKeys->end()) {
+        lockedOldEDNSCookieKeys->erase(it);
         return "COOKIE secret " + secret +" removed";
       }
     } catch (const std::exception& exc) {
@@ -190,11 +193,11 @@ string DLManageCookieSecret(const vector<string>& parts, [[maybe_unused]] pid_t
     return "COOKIE secret " + secret +" not found";
   }
   if (command == "list") {
-    if (DNSPacket::s_EDNSCookieKey.empty()) {
+    if (lockedEDNSCookieKey->empty()) {
       return "No EDNS Cookie secrets";
     }
-    oss << makeHexDump(DNSPacket::s_EDNSCookieKey, "") << " *";
-    for (const auto& secret : DNSPacket::s_OldEDNSCookieKeys) {
+    oss << makeHexDump(*lockedEDNSCookieKey, "") << " *";
+    for (const auto& secret : *lockedOldEDNSCookieKeys) {
       oss << std::endl << makeHexDump(secret, "");
     }
     return oss.str();
